elements/tags.lua

```diff
@@ -605,11 +605,42 @@ local tagPool = {}
 local funcPool = {}
 local tmp = {}
 
-local function getTagName(tag)
-	local tagStart = tag:match('>+()') or 2
-	local tagEnd = (tag:match('.-()<') or -1) - 1
+local function getBracketData(tag)
+	-- full tag syntax: '[prefix$>tag-name$<suffix(a,r,g,s){output-length}]'
+	local suffixEnd = (tag:match('()%(') or tag:match('(){') or -1) - 1
 
-	return tag:sub(tagStart, tagEnd), tagStart, tagEnd
+	local prefixEnd, prefixOffset = tag:match('()%$>'), 1
+	if(not prefixEnd) then
+		prefixEnd = 1
+	else
+		prefixEnd = prefixEnd - 1
+		prefixOffset = 3
+	end
+
+	local suffixStart, suffixOffset = tag:match('%$<()', prefixEnd), 1
+	if(not suffixStart) then
+		suffixStart = suffixEnd + 1
+	else
+		suffixOffset = 3
+	end
+
+	return tag:sub(prefixEnd + prefixOffset, suffixStart - suffixOffset), prefixEnd, suffixStart, suffixEnd, tag:match('%((.-)%)') or '', tonumber(tag:match('{(%d+)}'))
+end
+
+local function utf8shorten(str, length)
```

Comment:
We should take colour stuff like `|cffaabbcc` and `|r` into account here.

Reply:
Good catch, should also look out for `|T |t` and other similar things.